### ISSUE TITLE
Ruff fails to parse `WithItem` with a starred expression as `optional_vars`

### ISSUE THREAD

**Issue by LaBatata101** [astral-sh/ruff#8914]
Ruff version: v0.1.6

Ruff reports a syntax error for the following code
```python
with 0 as *x: pass
```
While the python `ast` module parses with no error.
```shell
$ python
Python 3.12.0 (main, Oct  3 2023, 01:27:23) [Clang 17.0.1 ] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import ast
>>> print(ast.dump(ast.parse("with a as *a: pass"), indent=4))
Module(
    body=[
        With(
            items=[
                withitem(
                    context_expr=Name(id='a', ctx=Load()),
                    optional_vars=Starred(
                        value=Name(id='a', ctx=Store()),
                        ctx=Store()))],
            body=[
                Pass()])],
    type_ignores=[])
>>>
```

[Playground](https://play.ruff.rs/6525a868-ddf2-46ff-99fe-5adc50f2dda8)

**Comment by MichaReiser** [comment]
Interesting. This does not parse for me in the REPL

```
>>> with 0 as *x: pass
...
  File "<stdin>", line 1
SyntaxError: starred assignment target must be in a list or tuple
```

**Comment by dhruvmanila** [comment]
> Interesting. This does not parse for me in the REPL

I think that might be a runtime error?

**Comment by zanieb** [comment]
Black throws a parse error for this too.

**Comment by dhruvmanila** [comment]
For context, here's where the error is being raised in CPython: https://github.com/python/cpython/blob/5b0629966f47542527400b03498d5156846f0da6/Python/compile.c#L6272-L6278

**Comment by dhruvmanila** [comment]
This is fixed with the new parser.
